pkg/codegen/json/README.md

```diff
@@ -0,0 +1,98 @@
+# PCL to JSON (experimental)
```

Comment:
> I am not sure JSON is the right approach

I feel like JSON is really accessible to everyone and can be easily consumed from any language without too much effort or knowledge.

>  it still requires us to code up classes to represent the JSON and parse it correctly.

I would argue that it is actually pretty straightforward to parse. Took me about an 1 - 2 hours to finish the whole thing. 

> I had been thinking of writing a protobuf format for this, given that program generation is eventually going to be on the grpc interface that had a nice synergy of us being able to send a structured.

My proposal here is that PCL-to-JSON is a good solution for people to experiment with PCL quickly using `pulumi convert --language json` without needing to wire gRPC clients over to the Pulumi CLI. Once they have a parser and feel comfortable wiring up with the Pulumi gRPC server, then they can reuse the parser if there is a gRPC method that returns that same JSON string.

Reply:
> I feel like JSON is really accessible to everyone and can be easily consumed from any language without too much effort or knowledge.

At the expense of static type information, and having a good way to keep that information in sync. Even JSON schema is a bit lackluster here because it's all too easy to write a schema where additional properties are inadvertently allowed and then emitting a new property not described in the schema but it still passing validation.
But if we go with the JSON route we should write a schema for it _and_ ensure we always check that our output is valid against that schema.

> without needing to wire gRPC clients over to the Pulumi CLI.
Protobuf doesn't require a grpc client. You can read/write binary (or json encoded) protobuf files directly.
And yeh given it already has a JSON format we could just use that for this, with the added benefit of "free" static types.

Reply:
> Even JSON schema is a bit lackluster here because it's all too easy to write a schema where additional properties are inadvertently allowed and then emitting a new property not described in the schema but it still passing validation.
But if we go with the JSON route we should write a schema for it and ensure we always check that our output is valid against that schema.


I think we can use `additionalProperties: false` to be more strict about it. I will write a JSON schema for the generated PCL and experiment with code generators 😄 

> Protobuf doesn't require a grpc client. You can read/write binary (or json encoded) protobuf files directly.
And yeh given it already has a JSON format we could just use that for this, with the added benefit of "free" static types.

I've never worked with the binary format so I don't know how developer-friendly it is. As for static types, I will experiment with a JSON schema that validates the PCL code